integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -498,3 +500,40 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
+
+def _spark_33008_datagen(spark_session: SparkSession) -> DataFrame:
```

Comment:
I'd like to see a more immediately descriptive name here rather than the indirect description via an Apache Spark JIRA ID.  Some code readers won't know that this is an Apache JIRA ID, and for those that do, they won't know what SPARK-33008 means without needing to look it up.  Ideally we should save both groups the frustration and describe the purpose of this dataset directly in its name.